### PR TITLE
chore: error-handling for log indexing

### PR DESCRIPTION
**Pull request by Omeramsc**
# Description

We encountered an edge-case in which the curl command that initiate the log indexing for Splunk silently fail. As the looping check we have later on only checks for indexing that already started, therefore ignoring what failed to begin.

This change will fail the script on the initiation of the log indexing, given an error will occur.

## Issue ticket number and link
None

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manual test:
```
[1/2] STEP 10/10: RUN bash -o errexit /opt/splunk/pre-run.sh
Indexing logs...
Indexing /opt/splunk/test_logs/log_example.log
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: 
Encountered an error while initiating indexing for /opt/splunk/test_logs/log_example.log. Exiting.
Error: error building at STEP "RUN bash -o errexit /opt/splunk/pre-run.sh": error while running runtime: exit status 1
```

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added meaningful description with JIRA/GitHub issue key (if applicable)
- [ ] I have updated labels (if needed)

### CODE REVIEW

**Review by amisstea — APPROVED** [review]

File: splunk/scripts/log_indexing.sh [hunk 1]
```diff
@@ -7,7 +7,10 @@ SPLUNK_PASSWORD="Password"
 echo "Indexing logs..."
 for file in "$logs_path"/*; do
   echo "Indexing $file"
-  curl  --insecure --user admin:"$SPLUNK_PASSWORD" "$endpoint" --data name="$file" --data index="test_index"
+  curl --insecure --user admin:"$SPLUNK_PASSWORD" "$endpoint" --data name="$file" --data index="test_index" || {
+    echo "Encountered an error while initiating indexing for $file. Exiting."
+    exit 1
```
Comment (ifireball) [inline splunk/scripts/log_indexing.sh:12]: Wouldn't it be more reliable to retry rather then just exit?

**Review by ifireball — CHANGES_REQUESTED** [review]

File: splunk/scripts/log_indexing.sh (hunk 1 above)
Comment (Omeramsc) [reply splunk/scripts/log_indexing.sh:12]: Yes, it is possible. I'll add it

File: splunk/scripts/log_indexing.sh [hunk 2]
```diff
@@ -5,9 +5,25 @@ endpoint="https://localhost:8089/services/data/inputs/oneshot"
...
 
 echo "Indexing logs..."
+max_retries=3
+
 for file in "$logs_path"/*; do
   echo "Indexing $file"
-  curl  --insecure --user admin:"$SPLUNK_PASSWORD" "$endpoint" --data name="$file" --data index="test_index"
+  retries=0
+
+  while [ $retries -lt $max_retries ]; do
```
Comment (ifireball) [inline splunk/scripts/log_indexing.sh:14]: please read up on the difference between the single and double-bracket syntax in bash, and try to use the newer syntax where possible.

Alternatively, we might be able to user `curl`s built-in `--retry` option.

File: splunk/scripts/log_indexing.sh [hunk 3]
```diff
@@ -5,9 +5,25 @@ endpoint="https://localhost:8089/services/data/inputs/oneshot"
...
+    if curl --insecure --user admin:"$SPLUNK_PASSWORD" "$endpoint" --data name="$file" --data index="test_index"; then
+      break
+    else
+      echo "Encountered an error while initiating indexing for $file. Retrying..."
+      retries=$((retries + 1))
+    fi
+
+    if [ $retries -eq $max_retries ]; then
```
Comment (ifireball) [inline splunk/scripts/log_indexing.sh:22]: can we somehow have more output here to help figuring out what happened?

**Review by ifireball — CHANGES_REQUESTED** [review]

File: splunk/scripts/log_indexing.sh (hunk 3 above)
Comment (Omeramsc) [reply splunk/scripts/log_indexing.sh:22]: stderr is printed to the console\log anyway, do we need more\duplicate output?

Comment (ifireball) [reply splunk/scripts/log_indexing.sh:22]: in your experience, was that sufficient to debug issues?

Comment (Omeramsc) [reply splunk/scripts/log_indexing.sh:22]:
```
Indexing logs...
Indexing /opt/splunk/test_logs/log_events.log
(7) Failed to connect to localhost port 8085: Connection refused
Encountered an error while initiating indexing for /opt/splunk/test_logs/log_events.log. Retrying...
```

Seems pretty sufficient to me... Is there anything you think should be added?

File: splunk/scripts/log_indexing.sh (hunk 2 above)
Comment (Omeramsc) [reply splunk/scripts/log_indexing.sh]: nice one, fixed

**Review by ifireball — APPROVED** [review]

**Pull request merged**